src/persist/benches/end_to_end.rs

```diff
@@ -77,6 +77,8 @@ fn write_and_bench_read<M: Measurement>(
     )
     .expect("error writing data");
 
+    runtime.stop().expect("runtime shut down cleanly");
```

Comment:
These are meant to make it explicit when the runtime is being shut down?

Reply:
This one actually moves the shut down earlier and is the motivation for this commit. (The one below is functionally the same and just makes it more explicit.)

Previously, the benchmark left this one running and then started a new one in `bench_read_persisted_source`, both pointed at the same storage. The storage allowed this since the reentrance_id matched, but the contract of the reentrance_id is that it's a user-enforced exclusivity token and the benchmark code was violating that contract. Shutting this one down before bench_read_persisted_source fixes that violation.

In practice, since nothing was being written to this first runtime, it happened to work and would eventually notice (via a debug-only assertion) that the META had changed from under it and would then panic.